docs: add a blog post for the new feature SSO (#836)
bytebase/bytebase.com: content/blog/introducing-single-sign-on-in-bytebase.md

---
title: Introducing Single Sign-On in Bytebase
author: Candy
published_at: 2023/03/07 17:37:17

> feature_image: _static_blog_introducing-single-sign-on-in-bytebase_cover.webp

Reviewer: the cover image is broken

Author: Already fixed it.

tags: Announcement
featured: true

> description: This article demonstrates the Single Sign-On protocols supported by Bytebase. With SSO, users log in once and access all authorized resources, resulting in improved security, user experience, and organizational flexibility.

Reviewer 2: ```suggestion
description: This article demonstrates the Single Sign-On (SSO) protocols supported by Bytebase. SSO allows users to log in once and access all authorized resources, improving security, user experience, and organizational flexibility.
```

---

## What’s Single Sign-On?

> Single Sign-On (SSO) is a user authentication service that allows users to access multiple applications or services with one set of login credentials. There are several benefits to supporting SSO:

Reviewer 2: ```suggestion
Single Sign-On (SSO) is a user authentication service allowing users to access multiple applications or services with one login credential. There are several benefits to supporting SSO:
```

## Supported Protocols in Bytebase

SSO is typically achieved through standardized authentication protocols such as SAML, LDAP, OAuth, or OpenID Connect (OIDC). Today, Bytebase supports the two most widely used protocols for SSO: [OAuth 2.0](https://www.rfc-editor.org/rfc/rfc6749) and [OIDC](https://openid.net/connect/).

### OAuth 2.0

OAuth 2.0, released in 2012 to replace OAuth 1.0, allows services or applications to access accounts on an HTTP service with limited permissions. With OAuth 2.0, user authentication is delegated to the service hosting the user account, and third-party applications are authorized to access that user account without sharing the user’s credentials.

Bytebase integrates the following identity providers supporting OAuth 2.0:

* GitHub
* GitLab
* Google
* Custom

For details information on how to configure SSO using OAuth 2.0, you can refer to the user doc [OAuth 2.0](/docs/administration/sso/oauth2).

### OpenID Connect (OIDC)

OIDC is an open standard for authentication and authorization that is built on top of OAuth 2.0. It provides a secure and reliable way for users to authenticate to an application using their existing online accounts.

> Like OAuth 2.0, OIDC relies on identity providers to authenticate users. These identity providers can be third-party services or self-hosted solutions. Byteabase provides the following identity providers for OIDC:

Reviewer 2: ```suggestion
Like OAuth 2.0, OIDC relies on identity providers to authenticate users. These identity providers can be third-party services or self-hosted solutions including:
```

Author: Bytebase just provides part of identity providers. So I prefer the original expression.
